frameworks/AutoGluon/exec.py

```diff
@@ -0,0 +1,108 @@
+import logging
+import os
+import warnings
+warnings.simplefilter("ignore")
+
+import matplotlib
+import pandas as pd
+matplotlib.use('agg')  # no need for tk
+
+from autogluon.task.tabular_prediction.tabular_prediction import TabularPrediction as task
+from autogluon.utils.tabular.utils.savers import save_pd, save_pkl
+import autogluon.utils.tabular.metrics as metrics
+
+from frameworks.shared.callee import call_run, result, Timer, touch
+
+log = logging.getLogger(__name__)
+
+
+def run(dataset, config):
+    log.info("\n**** AutoGluon ****\n")
+
+    metrics_mapping = dict(
+        acc=metrics.accuracy,
+        auc=metrics.roc_auc,
+        f1=metrics.f1,
+        logloss=metrics.log_loss,
+        mae=metrics.mean_absolute_error,
+        mse=metrics.mean_squared_error,
+        r2=metrics.r2
+    )
+
+    perf_metric = metrics_mapping[config.metric] if config.metric in metrics_mapping else None
+    if perf_metric is None:
+        # TODO: figure out if we are going to blindly pass metrics through, or if we use a strict mapping
+        log.warning("Performance metric %s not supported.", config.metric)
+
+    is_classification = config.type == 'classification'
+    training_params = {k: v for k, v in config.framework_params.items() if not k.startswith('_')}
+
+    train = pd.DataFrame(dataset.train.data, columns=dataset.columns)
```

Comment:
You mentioned [here](https://github.com/openml/automlbenchmark/pull/105#discussion_r423299073) that
```
pd.DataFrame(dataset.train.data, columns=dataset.columns)
```
Returns a DataFrame of all object types, but that we can get the true types through an edit to `__init__` and then:

```
train = pd.DataFrame(dataset.train.data, columns=data.columns).astype(column_types)
```

Where now train has types float and object. This would be ideal for AutoGluon.

Otherwise, AutoGluon will treat the floats as objects (It will not override the types given to it), dramatically reducing overall performance. This was why I originally would save the DataFrame to csv and reload it with AutoGluon, as that infers the types automatically and converts the floats to float types instead of being objects.

Reply:
hmm, there's possibly a flaw in the benchmark app then, that penalizes many sklearn-based frameworks...
The app only manipulates `numpy` arrays, so I can't set the type per column, and frameworks that can load files directly have this possibility.
If `amlb.data` module was exposing `pd.DataFrame` instead of `ndarray`, maybe this would globally tackle this concern for everyone..

Reply:
OK, I got a bit worried, but it's not that bad for sklearn-based frameworks (autosklearn, TPOT...): those are using encoded data which are typed as float.
So it's only a concern for frameworks that support categoricals, in which case I'll use the approach I recommended for mljar for now.